datacommons/test/set_api_key_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Data Commons Python API unit tests.
+
+Unit tests setting the API Key.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from unittest import mock
+
+import datacommons as dc
+import datacommons.utils as utils
+
+import json
+import unittest
+import urllib
+
+
+def request_mock(*args, **kwargs):
+  """ A mock urlopen call sent in the urllib package. """
+  # Create the mock response object.
+  class MockResponse:
+    def __init__(self, json_data):
+      self.json_data = json_data
+
+    def read(self):
+      return self.json_data
+
+  req = args[0]
+  data = json.loads(req.data)
+  # If the API key does not match, then return 403 Forbidden	
+  api_key = req.get_header('X-api-key')
+  if api_key != 'TEST-API-KEY':
+    return urllib.error.HTTPError(None, 403, None, None, None)
```

Comment:
this is interesting. so if setting a key, we validate, if not, pass through.

can you add a test to verify the alternate behavior, when a key isn't set? even if it's already tested implicitly elsewhere, would be nice to have an explicit test here.

Reply:
The validation is just to make sure what is set is what we specified. I've changed it to an environment var to test that what we set is what is sent. I also added a foobar before the final set to test this.

Adding a test for no set.